homeassistant/components/home_connect/sensor.py

```diff
@@ -38,47 +42,205 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+EVENT_OPTIONS_TRANSLATION_KEYS = {
+    BSH_EVENT_PRESENT_STATE_PRESENT: "present",
+    BSH_EVENT_PRESENT_STATE_CONFIRMED: "confirmed",
+    BSH_EVENT_PRESENT_STATE_OFF: "off",
+}
+
+
 @dataclass(frozen=True, kw_only=True)
 class HomeConnectSensorEntityDescription(SensorEntityDescription):
     """Entity Description class for sensors."""
 
-    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
-    options: list[str] | None = field(
-        default_factory=lambda: ["confirmed", "off", "present"]
-    )
-    desc: str
-    appliance_types: tuple[str, ...]
+    options_translation_keys: dict[str, str] | None = None
+    default_value: str | None = None
+    appliance_types: tuple[str, ...] | None = None
+    sign: int = 1
+
 
+BSH_PROGRAM_SENSORS = (
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.RemainingProgramTime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        sign=1,
+        translation_key="remaining_program_time",
+    ),
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.Duration",
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        sign=1,
+        translation_key="duration",
+    ),
```

Comment:
Add a device class

Reply:
Oh, okey! I didn't realised that there was a device class that fitted on that because it wasn't before at `apy.py`
https://github.com/home-assistant/core/blob/0d795aad16fa095f93f305e428ec26d7dadf93e6/homeassistant/components/home_connect/api.py#L199-L205